src/raftstore/store/peer.rs

```diff
@@ -1203,24 +1225,69 @@ impl Peer {
         self.proposals.gc();
     }
 
+    /// Responses to the ready read request.
+    fn post_pending_reads<T, C>(&mut self, ctx: &mut PollContext<T, C>) {
+        if self.pending_reads.ready_cnt > 0 {
+            for _ in 0..self.pending_reads.ready_cnt {
+                let (read_index, is_read_index_request) = {
+                    let read = self.pending_reads.reads.front().unwrap();
+                    if read.cmds.len() == 1
+                        && read.cmds[0].0.get_requests().len() == 1
+                        && read.cmds[0].0.get_requests()[0].get_cmd_type() == CmdType::ReadIndex
+                    {
+                        (read.read_index, true)
+                    } else {
+                        (read.read_index, false)
+                    }
+                };
+
+                if !self.ready_to_handle_read() && !is_read_index_request {
+                    break;
+                }
+                let mut read = self.pending_reads.reads.pop_front().unwrap();
+                for (req, cb) in read.cmds.drain(..) {
+                    cb.invoke_read(self.handle_read(ctx, req, true, read_index));
+                }
+                self.pending_reads.ready_cnt -= 1;
+            }
+        }
+    }
+
     fn apply_reads<T, C>(&mut self, ctx: &mut PollContext<T, C>, ready: &Ready) {
         let mut propose_time = None;
+        // The follower may lost `ReadIndexResp`, so the pending_reads is not
```

Comment:
A network may also drop `ReadIndexRequest` silently, should we retry read index after few heartbeats timeouts?

Reply:
If the next request succeeds，the before request also can be processed even which may be lost in `Ready`. and the client better set the timeout for the request.